webui/src/lib/hooks/repo.jsx

```diff
@@ -28,7 +28,7 @@ export const resolveRef = async (repoId, refId) => {
         const commit = await commits.get(repoId, refId);
         return {id: commit.id,  type: 'commit'};
     } catch(error) {
-        if (!(error instanceof NotFoundError)) {
+        if (!error instanceof NotFoundError && !error instanceof BadRequestError) {
```

Comment:
In case we've reached here and ref is not a branch, tag or a commit and the error is aBadRequestError - shouldn't we throw the actual error instead of throwing a NotFoundError in line 36

Reply:
Done :)